rules/prevent-errortype.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+const {visitIf} = require('enhance-visitors');
+const util = require('../util');
+const createAvaRule = require('../create-ava-rule');
+
+const nameRegexp = /^(?:[A-Z][a-z0-9]*)*Error$/;
+
+const create = context => {
+	const ava = createAvaRule();
+
+	return ava.merge({
+		CallExpression: visitIf([
+			ava.isInTestFile,
+			ava.isInTestNode
+		])(node => {
+			const functionArgIndex = node.arguments.length - 1;
+			let functionArgName;
+
+			if (typeof node.callee.property === 'undefined') {
+				return;
+			}
+
+			const calleeProperty = node.callee.property.name;
+
+			if (functionArgIndex === 1) {
+				functionArgName = node.arguments[1].name;
+			} else {
+				return;
+			}
```

Comment:
Would be clearer to invert the if-condition and use a early-return here.

Reply:
done.